src/emit.ts

```diff
@@ -10,43 +11,44 @@ let events: Record<string, typeof componentEvents>
 export function emitted<T = unknown>(
   vm: ComponentPublicInstance,
   eventName?: string
-): T[] | Record<string, T[]> {
+): undefined | T[] | Record<string, T[]> {
   const cid = vm.$.uid
 
   const vmEvents = (events as Record<string, Record<string, T[]>>)[cid] || {}
   if (eventName) {
-    const emitted = vmEvents
-      ? (vmEvents as Record<string, T[]>)[eventName]
-      : undefined
-    return emitted
+    return vmEvents ? (vmEvents as Record<string, T[]>)[eventName] : undefined
   }
 
   return vmEvents as Record<string, T[]>
 }
+type Events = { [id: number]: Record<string, any> }
 
 export const attachEmitListener = () => {
   events = {}
   // use devtools to capture this "emit"
   setDevtoolsHook(createDevTools(events))
 }
 
-function createDevTools(events): any {
-  const devTools: Partial<typeof devtools> = {
+function createDevTools(events: Events): any {
```

Comment:
instead of `as Partial` should we just change `: any` to `: Partial<typeof devtools>`

Reply:
unfortunatly not, `setDevtoolsHook` expects a full [DevtoolsHook](https://github.com/vuejs/vue-next/blob/3626ff07fe5107080c52e85018070562c84b796e/packages/runtime-core/src/devtools.ts#L21) implementation which partial doesn't cover. I'm not sure what's the logic should be for the rest. I can add stub implementations.